mentat/session_input.py

```diff
@@ -1,25 +1,46 @@
 import asyncio
 import logging
+import shlex
 from typing import Any, Coroutine
 
-from .errors import RemoteKeyboardInterrupt
+from .errors import RemoteKeyboardInterrupt, SessionExit
 from .session_stream import SESSION_STREAM, StreamMessage
 
 
-async def collect_user_input(**kwargs: Any) -> StreamMessage:
+async def _get_input_request(**kwargs: Any) -> StreamMessage:
+    stream = SESSION_STREAM.get()
+
+    message = await stream.send("", channel="input_request", **kwargs)
+    response = await stream.recv(f"input_request:{message.id}")
+    return response
+
+
+async def collect_user_input(plain: bool = False) -> StreamMessage:
     """Listens for user input on a new channel
 
     send a message requesting user to send a response
     create a new broadcast channel that listens for the input
     close the channel after receiving the input
     """
-    stream = SESSION_STREAM.get()
-
-    message = await stream.send("", channel="input_request", **kwargs)
-    response = await stream.recv(f"input_request:{message.id}")
+    # This fixes a circular import
+    from mentat.commands import Command
```

Comment:
This also makes me want to have a single `SESSION_CONTEXT` lol

Reply:
🙃